ote_sdk/ote_sdk/usecases/exportable_code/demo/demo_package/executors/synchronous.py

```diff
@@ -28,8 +31,68 @@ class SyncExecutor:
 
     def __init__(self, model: ModelContainer, visualizer: IVisualizer) -> None:
         self.model = model.core_model
+        self.params = model.parameters
         self.visualizer = visualizer
         self.converter = create_output_converter(model.task_type, model.labels)
+        self.task_type = model.task_type
+        self.tiler = self.setup_tiler()
+
+    def setup_tiler(self):
+        """Setup tiler
+
+        Returns:
+            Tiler: tiler module
+        """
+        if (
+            not self.params.get("tiling_parameters")
+            or not self.params["tiling_parameters"]["enable_tiling"]["value"]
+        ):
+            return None
+
+        segm = False
+        if (
+            self.task_type is TaskType.ROTATED_DETECTION
+            or self.task_type is TaskType.INSTANCE_SEGMENTATION
+        ):
+            segm = True
+        tile_size = self.params["tiling_parameters"]["tile_size"]["value"]
+        tile_overlap = self.params["tiling_parameters"]["tile_overlap"]["value"]
+        max_number = self.params["tiling_parameters"]["tile_max_number"]["value"]
+        tiler = Tiler(tile_size, tile_overlap, max_number, self.model, segm)
+        return tiler
+
+    def infer(self, frame):
+        """Infer with original image
+
+        Args:
+            frame (np.ndarray): image
+
+        Returns:
+            annotation_scene (AnnotationScene): prediction
+            frame_meta (Dict): dict with original shape
+        """
+        # getting result include preprocessing, infer, postprocessing for sync infer
+        predictions, frame_meta = self.model(frame)
+        predictions = detection2array(predictions)
+        annotation_scene = self.converter.convert_to_annotation(predictions, frame_meta)
+        return annotation_scene, frame_meta
+
+    def infer_tile(self, frame):
+        """Infer by patching full image to tiles
+
+        Args:
+            frame (np.ndarray): image
+
+        Returns:
+            annotation_scene (AnnotationScene): prediction
+            frame_meta (Dict): dict with original shape
+        """
+
+        detections, _ = self.tiler.predict(frame)
+        annotation_scene = self.converter.convert_to_annotation(
+            detections, metadata={"original_shape": frame.shape}
+        )
+        return annotation_scene, None
```

Comment:
I don't think this is the best place to put this tiling & branching logic. What do you think of hiding this to model container at least?
In addition, I don't see many differences from OpenVINOXXXXInferencer in openvino_task.py.
Can we remove code or logic duplications here & there?

For example, we can move all "BaseInferencer" variants in detection OV tasks to SDK, then use them both in OV task and exportable code demo.

Reply:
* I moved the logic to model container
* I have added a ticket: CVS-98096 tracking this refactoring work